### PR TITLE
[# 157298166] As a developer, so I can run rubocop locally without errors, update rubocops configs to comply with codeclimates version of 0.52.1

### PR DESCRIPTION
**Pull request by amulder87**
As a developer, so I can run rubocop locally without errors, update rubocops configs to comply with codeclimates version of 0.52.1

[Tracker #157298166](https://www.pivotaltracker.com/story/show/157298166)

### CODE REVIEW

File: ruby/rails/rubocop.yml [hunk 1]
```diff
@@ -289,48 +289,38 @@ Style/EmptyElse:
...
 
-# Checks whether the source file has a utf-8 encoding comment or not
-# AutoCorrectEncodingComment must match the regex
-# /#.*coding\s?[:=]\s?(?:UTF|utf)-8/
-Style/Encoding:
-  EnforcedStyle: always
-  SupportedStyles:
-    - when_needed
-    - always
-  AutoCorrectEncodingComment: '# encoding: utf-8'
```
Comment (amulder87) [inline ruby/rails/rubocop.yml:323]: The UTF-8 encoding became default in ruby 2.0.0 and rendered the magic comment for it obsolete. Rubocop now enforces the "never" style by default.

File: ruby/rails/rubocop.yml [hunk 2]
```diff
@@ -412,10 +402,7 @@ Style/HashSyntax:
 # Force hashes that have a symbol value to use hash rockets
   UseHashRocketsWithSymbolValues: false
 
-Style/IfUnlessModifier:
-  MaxLineLength: 80
```
Comment (amulder87) [inline ruby/rails/rubocop.yml:416]: Obsolete, governed by Metrics/LineLength: Max

File: ruby/rails/rubocop.yml [hunk 3]
```diff
@@ -800,15 +787,12 @@ Style/TrivialAccessors:
...
 
-Style/VariableName:
+Naming/VariableName:
   EnforcedStyle: snake_case
   SupportedStyles:
     - snake_case
     - camelCase
 
-Style/WhileUntilModifier:
-  MaxLineLength: 80
```
Comment (amulder87) [inline ruby/rails/rubocop.yml:810]: Obsolete, governed by Metrics/LineLength: Max

**Review by RoryO — APPROVED** [review]

**Comment by amulder87** [comment]
@RoryO I was getting strange behavior still after making updates to the configs. It looks like it's because the inheritance in the config was being applied in reverse order. The `inherit_from` in `rubocop.yml` was trying to use `rubocop-overrides.yml` and `rubocop-disabled.yml` to override the base config in `rubocop.yml`. But, it works the other way around, so the overrides and disableds were acting as the base config, and had their settings overriden by the defaults in `rubocop.yml`.

I'm not sure if this approach is the best way to fix this though, because it means needing to update the .codeclimate.yml in every repository that uses it to add the extra `rubocop-base.yml` to the `codeclimate prepare` fetch configuration.

**Comment by amulder87** [comment]
@RoryO Plan for this now would be having just a `rubocop-defaults.yml`(representing a stock config for rubocop) and a `rubocop-overrides.yml`(representing our org-wide overrides of stock stettings) in `code-standards`. Every repo that runs rubocop based off the centralized config would have its own git-tracked `.rubocop.yml` that inherits from the centralized config and can includes its own overrides.

After this PR, I would open PRs for every necessary repo to update their `.codeclimate.yml`s to fetch just these 2 configs, update `.gitignore`s to ignore just the defaults/overrides and no longer ignore `.rubocop.yml` and to add a basic `.rubocop.yml` that could be updated per repo that would look like this:

```
# inherit centralized default configuration
inherit_from:
  - .rubocop-defaults.yml
  - .rubocop-overrides.yml

# add any local overrides below
```

Does that all sound good?

**Pull request merged**